src/main/java/com/minecolonies/coremod/network/messages/server/colony/building/university/TryResearchMessage.java

```diff
@@ -76,50 +96,132 @@ protected void onExecute(
+                            player.sendMessage(new TranslationTextComponent("com.minecolonies.coremod.research.requirementnotmet"), player.getUniqueID());
+                            return;
+                        }
+                    }
+                    // Remove items from player
+                    if (!InventoryUtils.tryRemoveStackFromItemHandler(new InvWrapper(player.inventory), research.getCostList()))
+                    {
+                        player.sendMessage(new TranslationTextComponent("com.minecolonies.coremod.research.costnotavailable", new TranslationTextComponent(research.getDesc())),
+                          player.getUniqueID());
+                        return;
                     }
-
-                    player.sendMessage(new TranslationTextComponent("com.minecolonies.coremod.research.started", research.getDesc()), player.getUniqueID());
-                    research.startResearch(player, colony.getResearchManager().getResearchTree());
                 }
-                colony.markDirty();
-                // Remove items from player
+                player.sendMessage(new TranslationTextComponent("com.minecolonies.coremod.research.started", new TranslationTextComponent(research.getDesc())),
+                  player.getUniqueID());
+                research.startResearch(colony.getResearchManager().getResearchTree());
             }
         }
         else
         {
-            if (player.isCreative() && colony.getResearchManager().getResearchTree().getResearch(branch, researchId).getState() == ResearchState.IN_PROGRESS)
+            if(player.isCreative())
             {
-                final IGlobalResearch research = IGlobalResearchTree.getInstance().getResearch(branch, researchId);
-                colony.getResearchManager().getResearchTree().getResearch(branch, research.getId()).setProgress((int) (BASE_RESEARCH_TIME * Math.pow(2, research.getDepth() - 1)));
+                if (MinecoloniesAPIProxy.getInstance().getConfig().getServer().researchCreativeCompletion.get())
+                {
+                    colony.getResearchManager()
+                      .getResearchTree()
+                      .getResearch(branch, research.getId())
+                      .setProgress((int) (BASE_RESEARCH_TIME * Math.pow(2, research.getDepth() - 1)));
+                }
             }
             else
             {
                 player.sendMessage(new TranslationTextComponent("com.minecolonies.coremod.research.alreadystarted"), player.getUniqueID());
             }
         }
+        colony.markDirty();
+    }
+
+    /**
+     * Attempt to reset research for a colony.
+     * @param player     the player making the request (and to apply costs toward)
+     * @param colony     the colony to remove the research from.
+     * @param research   the research.
+     */
+    private void attemptResetResearch(final PlayerEntity player, final IColony colony, final IGlobalResearch research)
```

Comment:
I feel like this code should be in the ResearchManager or even in the ResearchTree